contracts/swap/swap.go

```diff
@@ -97,6 +103,25 @@ func InstanceAt(address common.Address, backend Backend) (Contract, error) {
 	return c, err
 }
 
+// Withdraw withdraws amount from the chequebook and blocks until the transaction is mined
+func (s simpleContract) Withdraw(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error) {
+	tx, err := s.instance.Withdraw(auth, amount)
+	if err != nil {
+		return nil, err
+	}
+	return WaitFunc(auth, backend, tx)
+}
+
+// Deposit sends a transaction to the chequebook, which deposits the amount set in Auth.Value and blocks until the transaction is mined
+func (s simpleContract) Deposit(auth *bind.TransactOpts, backend Backend) (*types.Receipt, error) {
+	rawSimpleSwap := contract.SimpleSwapRaw{Contract: s.instance}
+	tx, err := rawSimpleSwap.Transfer(auth)
```

Comment:
just out of curiosity... why can `Withdraw` be called from `simpleContract.instance`, yet `Deposit` must be called through a `rawSimpleSwap` variable?

is this because one is a read-only operation?

Reply:
`Deposit` is not a function which is exposed by the smart-contract. In fact, deposit is a call with no `calldata`, but with `value`. Because no `calldata` doesn't match any of the defined function signatures,  the `fallback` function is triggered [see sw-3](https://github.com/ethersphere/swap-swear-and-swindle/blob/2aec7a67458dd8100e635576299ca3c544800698/contracts/SimpleSwap.sol#L258). Because there is `value` in the call, the smart contract will emit a `Deposit` event.
Abigen does not expose the fallback function by default, which is why I had to make this small hack to be still able to call the fallback.

Reply:
Please tell me if that was clear ;)

Reply:
clear enough! thanks